pydicom/jsonrep.py

```diff
@@ -41,3 +47,202 @@ def convert_to_python_number(value, vr):
+                fmt = '"{}" of data element "{}" must be a list.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            if not self.value:
+                return empty_value_for_VR(self.vr)
+            element_value = []
+            for v in self.value:
+                element_value.append(self.get_regular_element_value(v))
+            if len(element_value) == 1 and self.vr != 'SQ':
+                element_value = element_value[0]
+            return convert_to_python_number(element_value, self.vr)
+
+        # The value for "InlineBinary" shall be encoded as a base64 encoded
+        # string, as shown in PS3.18, Table F.3.1-1, but the example in
+        # PS3.18, Annex F.4 shows the string enclosed in a list.
+        # We support both variants, as the standard is ambiguous here,
+        # and do the same for "BulkDataURI".
+        value = self.value
+        if isinstance(value, list):
+            value = value[0]
+
+        if self.value_key == 'InlineBinary':
+            if not isinstance(value, compat.char_types):
+                fmt = '"{}" of data element "{}" must be a bytes-like object.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            return base64.b64decode(value)
+
+        if self.value_key == 'BulkDataURI':
+            if not isinstance(value, compat.string_types):
+                fmt = '"{}" of data element "{}" must be a string.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            if self.bulk_data_uri_handler is None:
+                warnings.warn(
+                    'no bulk data URI handler provided for retrieval '
+                    'of value of data element "{}"'.format(self.tag)
+                )
+                return empty_value_for_VR(self.vr, raw=True)
+            return self.bulk_data_uri_handler(value)
+        return empty_value_for_VR(self.vr)
+
+    def get_regular_element_value(self, value):
+        """Return a the data element value created from a json "Value" entry.
+
+        Parameters
+        ----------
+        value : str or int or float or dict
+            The data element's value from the json entry.
+
+        Returns
+        -------
+        dataset_class or PersonName3 or PersonNameUnicode
+        or str or int or float
+            A single value of the corresponding :class:`DataElement`.
+        """
+        if self.vr == 'SQ':
+            return self.get_sequence_item(value)
+
+        if self.vr == 'PN':
+            return self.get_pn_element_value(value)
+
+        if self.vr == 'AT':
+            try:
+                return int(value, 16)
+            except ValueError:
+                warnings.warn('Invalid value "{}" for AT element - '
+                              'ignoring it'.format(value))
+            return
+        return value
+
+    def get_sequence_item(self, value):
+        """Return a sequence item as a `dataset_class` instance.
```

Comment:
`dataset_class` isn't a parameter

Reply:
Ah, forgot to adapt the comment, thanks!